### PR TITLE
Exceptions Exercise

### PR DESCRIPTION
**Pull request by Iansberg**
try:
    Exercise
except pull_request:
    print('Thanks!')

### CODE REVIEW

File: students/Sahlberg/Lesson5/mailroom3/mailroom3.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                amount_append(option, donor, amount)
+                email(option,amount)
+
+    while user_action == '2':
+        donor_sorted = sort_and_stats_dict(donor)
+        print_report(donor_sorted)
+        break
+
+    while user_action == '3':
+        email_file(donor)
```
Comment (natasha-aleksandrova) [inline students/Sahlberg/Lesson5/mailroom3/mailroom3.py]: you should be using a switch dict pattern here (from Lesson 4) with exception handling.

File: students/Sahlberg/Lesson5/mailroom3/mailroom3.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
+"""Ian Sahlberg
+Assignment 5 mailroom3
+Python 210
+01/14/2019"""
+
+#for file dating
+import datetime as dt
+
+#Dictionary of donors name, dollar amount
+donor = {'bob johnson': [150.00],
```
Comment (natasha-aleksandrova) [inline students/Sahlberg/Lesson5/mailroom3/mailroom3.py]: should be `donors`

File: students/Sahlberg/Lesson5/mailroom3/mailroom3.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    """Generates a thank you letter and writes to disk for all donors."""
+    count = 0
+    count_file = 0
+    for name in dictionary.keys():
+        amount = [sum(v) for v in dictionary.values()][count]
+        file_name = '{}_{}.txt'.format(name, dt.date.today())
+        count += 1
+        with open(file_name, 'w+') as file:
+            e_text = str(email(name, amount))
+            count_file += 1
```
Comment (natasha-aleksandrova) [inline students/Sahlberg/Lesson5/mailroom3/mailroom3.py]: what is the purpose of this var?

File: students/Sahlberg/Lesson5/mailroom3/mailroom3.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            f'{dictionary[count][0]:<20}$    {dictionary[count][1][0]:<23.2f}{dictionary[count][1][1]:<15}    ${dictionary[count][1][2]:>12.0f}')
+        count += 1
+
+def email_file(dictionary):
+    """Generates a thank you letter and writes to disk for all donors."""
+    count = 0
+    count_file = 0
+    for name in dictionary.keys():
+        amount = [sum(v) for v in dictionary.values()][count]
+        file_name = '{}_{}.txt'.format(name, dt.date.today())
```
Comment (natasha-aleksandrova) [inline students/Sahlberg/Lesson5/mailroom3/mailroom3.py]: this is not quite correct pattern, see below:

```
for name, donations in donors.items():
    total = sum(donations)
```
above will you give you your "amount" - what you're doing currently is loop over your *whole* dictionary each time when the loop already gives you an element in your dict.

File: students/Sahlberg/Lesson5/mailroom3/mailroom3.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+def print_report(dictionary):
+    """Print out a report of donors and amounts donated, including stats (total donated, number of times donated, average donation)."""
+    print('Donor Name       |      Donor Total    |  Count of Donations    |  Average of Donations')
+    print('----------------------------------------------------------------------------------------')
+    count = 0
+    for donor in dictionary:
+        print(
+            f'{dictionary[count][0]:<20}$    {dictionary[count][1][0]:<23.2f}{dictionary[count][1][1]:<15}    ${dictionary[count][1][2]:>12.0f}')
+        count += 1
```
Comment (natasha-aleksandrova) [inline students/Sahlberg/Lesson5/mailroom3/mailroom3.py]: a bit confused here - why is dictionary accessed by index? loop over dictionary `for key, value in a_dict` to get key and value right away.

**Comment by Iansberg** [comment]
I will take a look at these comments.

 I think I'm not understanding the flow of submitting. I am not done with
lesson five assignments and didn't submit anything in the class website.
Maybe the issue is that I'm working on several assignments at once. My
mailroom two hadnt been graded, maybe still isn't.

Ian

**Comment by natasha-aleksandrova** [comment]
@Iansberg any commits with an open pull request will show under that pull request - that's probably where the confusion is, and since some students forget to submit on the course website it may get confusing.

well you got early comments 🙂 that you can apply to your actual submission

**Comment by Iansberg** [comment]
Lucky me! Switch case is now in my head. Instead of if if if use a
dictionary. Check.

**Pull request merged**